sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DescribeTableExec.scala

```diff
@@ -80,13 +81,25 @@ case class DescribeTableExec(
   }
 
   private def addPartitioning(rows: ArrayBuffer[InternalRow]): Unit = {
-    rows += emptyRow()
-    rows += toCatalystRow("# Partitioning", "", "")
-    if (table.partitioning.isEmpty) {
-      rows += toCatalystRow("Not partitioned", "", "")
-    } else {
-      rows ++= table.partitioning.zipWithIndex.map {
-        case (transform, index) => toCatalystRow(s"Part $index", transform.describe(), "")
+    if (table.partitioning.nonEmpty) {
+      val partitionColumnsOnly = table.partitioning.forall(t => t.isInstanceOf[IdentityTransform])
+      if (partitionColumnsOnly) {
+        rows += toCatalystRow("# Partition Information", "", "")
+        rows += toCatalystRow(s"# ${output(0).name}", output(1).name, output(2).name)
+        val nameToField = table.schema.map(f => (f.name, f)).toMap
+        rows ++= table.partitioning
+          .map(_.asInstanceOf[IdentityTransform])
+          .flatMap(_.ref.fieldNames())
```

Comment:
I think it's more tricky here, as the reference can be a nested field, e.g. `a.b`. We can still keep the output in a v1 compatible way, but the code to find its type and comment will be a bit more complicated, as we need to use `StructType.findNestedField`